tests/unit/limitations_test.c

```diff
@@ -457,3 +509,35 @@ create_default_cfg(splinterdb_config *out_cfg, data_config *default_data_cfg)
                           .extent_size = TEST_CONFIG_DEFAULT_EXTENT_SIZE,
                           .data_cfg    = default_data_cfg};
 }
+
+/*
+ * Helper function to parse command-line arguments to setup the configuration
+ * for SplinterDB.
+ */
+static platform_status
+parse_cmdline_args(void *datap, int unit_test_argc, char **unit_test_argv)
+{
+   // Cast void * datap to ptr-to-CTEST_DATA() struct in use.
+   struct CTEST_IMPL_DATA_SNAME(limitations) *data =
+      (struct CTEST_IMPL_DATA_SNAME(limitations) *)datap;
```

Comment:
Rant:

I know we use this construction across tests and is out of scope, but I hate this construction everywhere in the tests. I don't understand what is good or useful about it, and it all violates both the naming conventions and the underlying philosophy. We don't use `XXXp` for pointers, no one knows what a `struct CTEST_IMPL_DATA_SNAME` is supposed to be, it isn't a name that we would use, and a newcomer could not possibly infer what is happening here. I continue to hate it and I think we should change how this works.

Reply:
Brief reason: This is something we inherited from the CTests framework, which synthesizes structures to encapsulate data structures declared in each tests' `CTEST_SETUP()`, using the test's name [i.e., say here, `limitations`] as a sub-part of the generated structure's name.

The reason why this messy stuff is here is because we want to pass-down an argument to `<Generated-struct's-name> * ptr` to a sub-function which wants to extract out individual fields. Otherwise, the interface for these minions will needlessly grow, having to explicitly list out ptrs to individual `data->member` fields.

`CTEST_IMPL_DATA_SNAME()` is a macro defined in ctests `unit/ctest.h` file which evaluates to the generated name of a C-struct that is produced by the CTests machinery.

I understand your concern about names like `XXXp` for pointers, and that can be changed easily.

I suggest you & I talk off-line, converge on an acceptable solution. And then I will log a cleanup issue to implement the corrected approach in all unit-tests.